lasagne/layers/recurrent.py

```diff
@@ -107,7 +107,12 @@ class CustomRecurrentLayer(MergeLayer):
         :class:`lasagne.layers.Layer` instance which connects input to the
         hidden state (:math:`f_i`).  This layer may be connected to a chain of
         layers, which must end in a :class:`lasagne.layers.InputLayer` with the
-        same input shape as `incoming`.
+        same input shape as incoming, except for the first dimension: When
+        ``precompute_input == True`` (the default),
+        ``input_to_hidden.input_shape[0]`` must be
```

Comment:
> If you want to change it or make other fixes, maybe it would be faster for you to make them then squash/merge?

Hmm, I don't like pushing something to master that's not reviewed... but we can discuss the formulation here instead of going via commits. So my first proposal was:
```
with the same input shape as `incoming`, except for the first dimension:
That must be ``None``,
or ``incoming.output_shape[0]*incoming.output_shape[1]`` if ``precompute_input=True``,
or ``incoming.output_shape[0]`` if ``precompute_input=False``.
```

If that reads confusing for you, what about putting `None` in the end:
```
with the same input shape as `incoming`, except for the first dimension:
That must be ``incoming.output_shape[0]*incoming.output_shape[1]``
if ``precompute_input=True`` (the default), or ``incoming.output_shape[0]``
if ``precompute_input=False``, or simply left unspecified as ``None``.
```